pubsub/gcloud/aio/pubsub/subscriber.py

```diff
@@ -99,6 +100,20 @@ async def acker(subscription: str,
                 log.warning(
                     'Ack request failed, better luck next batch', exc_info=e)
                 metrics_client.increment('pubsub.acker.batch.failed')
+
+                is_response_error = isinstance(
+                    e,
+                    aiohttp.client_exceptions.ClientResponseError)
+                if is_response_error and e.status == 400:  # type: ignore # pylint: disable=no-member
```

Comment:
I think this would be cleaner by doing `except aiohttp.client_exceptions.ClientResponseError` as a fully different branch -- it would lead to duplicating the log warning and metrics, but I think that's OK -- we fundamentally want to react to the different exception type in a different way, so that's not the sort of duplication which I consider to be a bad thing.

Alternatively, you could wrap the exceptions at a different level, eg.
```python
try:
  try:
    await subscriber_client.acknowledge(..)
  except aiohttp.client_exceptions.ClientResponseError as e:
    if e.status == 400:
      # SNIP: your new code here
    raise
except Exception:
  log.warning(..)
  metrics_client.increment(..)
```